### PR TITLE
Use `Report` and `ReportBody` from `mc-sgx-core-types`

### PR DESCRIPTION
**Pull request by nick-mobilecoin**
Previously the `Report` and `ReportBody` were defined in
`mc-attest-core`. Now the `Report` and `ReportBody` from
`mc-sgx-core-types` is used.

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* master
  * **PR #3434** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/mobilecoin/3434" target="_blank">Graphite</a> 
    * **PR #3435** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/mobilecoin/3435" target="_blank">Graphite</a> 
      * **PR #3438** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/mobilecoin/3438" target="_blank">Graphite</a> 
        * **PR #3439** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/mobilecoin/3439" target="_blank">Graphite</a> 
          * **PR #3440** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/mobilecoin/3440" target="_blank">Graphite</a> 
            * **PR #3441** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/mobilecoin/3441" target="_blank">Graphite</a> 
              * **PR #3442** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/mobilecoin/3442" target="_blank">Graphite</a> 
                * **PR #3443** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/mobilecoin/3443" target="_blank">Graphite</a> 
                  * **PR #3444** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/mobilecoin/3444" target="_blank">Graphite</a> 
                    * **PR #3445** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/mobilecoin/3445" target="_blank">Graphite</a> 
                      * **PR #3449** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/mobilecoin/3449" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/mobilecoin/3449?utm_source=stack-comment).

File: attest/core/src/quote.rs [hunk 1]
```diff
@@ -333,6 +332,61 @@ impl Quote {
 
         Ok(())
     }
+
+    fn verify_report_body(
```
Comment (nick-mobilecoin) [inline attest/core/src/quote.rs:336]: This logic is pretty much copied from, https://github.com/mobilecoinfoundation/mobilecoin/blob/36be4bf1773ecb302c2204f09a3a15563da21333/attest/core/src/types/report_body.rs#L144

File: attest/enclave-api/src/lib.rs [hunk 2]
```diff
@@ -103,7 +103,7 @@ pub struct PlaintextClientRequest {
 /// The response to a request for a new report. The enclave will expect the
 /// QuoteNonce to be used when the report is quoted, and both the quote and
 /// report to be returned to the enclave during the verify_quote() phase.
-#[derive(Clone, Debug, Deserialize, Hash, Eq, Ord, PartialEq, PartialOrd, Serialize)]
+#[derive(Clone, Debug, Deserialize, Hash, Eq, PartialEq, Serialize)]
```
Comment (nick-mobilecoin) [inline attest/enclave-api/src/lib.rs:106]: @eranrund This is the higher level struct that wanted Ord and PartialOrd.

I'm not opposed to adding them back to the types, I mainly removed them as they didn't appear to be needed and they weren't implemented in `mc-sgx-core-types`

**Review by eranrund — APPROVED** [review]

**Comment by nick-mobilecoin** [comment]
@nick-mobilecoin started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/mobilecoin/3449).

**Comment by nick-mobilecoin** [comment]
[Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/mobilecoin/3449) couldn't merge this pull request because a downstack PR #3442 failed to merge.

**Comment by nick-mobilecoin** [comment]
@nick-mobilecoin started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/mobilecoin/3449).

**Comment by nick-mobilecoin** [comment]
[Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/mobilecoin/3449) rebased this pull request as part of a merge.

**Comment by nick-mobilecoin** [comment]
@nick-mobilecoin merged this pull request with [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/mobilecoin/3449).

**Pull request merged**